boolean_index.py

```diff
@@ -57,6 +57,45 @@ def build_cacm(self, filename: str):
 
         self.incidence_matrix = self.incidence_matrix.tocsc()  # Faster column slicing
 
+    def build_cs276(self,directory_name : str, from_saved_matrix : bool):
+        """
+        Uses the CS276 (Stanford) collection to build the class attributes
+
+        Args :
+            directory_name : String containing the path to the CS276 dataset (pa1-data)
+            from_saved_matrix  : True in order not to read again the whole dataset and load a saved matrix
+        """
+        doc_id = -1
+        term_id = -1
+
+        nb_terms = 353975
+        nb_docs = 98998
+        self.incidence_matrix = lil_matrix((nb_terms, nb_docs))
+
+        for block_id in range(10) :
+            list_of_files = os.listdir(os.path.join(directory_name, str(block_id)))
+            for filename in list_of_files:
+                file = open(os.path.join(directory_name, str(block_id), filename), "r")
```

Comment:
Tu devrais penser à [fermer le fichier](https://stackoverflow.com/questions/8011797/open-read-and-close-a-file-in-1-line-of-code), peut-être que ça accélérerait les choses mais pas sûr.